core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -5775,73 +5845,61 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                         PUT_V128_TO_ADDR(frame_lp + addr_ret, LOAD_V128(maddr));
                         break;
                     }
-#define SIMD_LOAD_OP(op_name, simde_func, element_size, num_elements)          \
-    do {                                                                       \
-        uint32 offset, addr;                                                   \
-        offset = read_uint32(frame_ip);                                        \
-        addr = GET_OPERAND(uint32, I32, 0);                                    \
-        frame_ip += 2;                                                         \
-        addr_ret = GET_OFFSET();                                               \
-        CHECK_MEMORY_OVERFLOW(16);                                             \
-                                                                               \
-        simde_v128_t simde_result = simde_func(maddr);                         \
-                                                                               \
-        V128 result;                                                           \
-        SIMDE_V128_TO_SIMD_V128(simde_result, result);                         \
-                                                                               \
-        V128 reversed_result;                                                  \
-        for (int i = 0; i < num_elements; i++) {                               \
-            reversed_result.i##element_size##x##num_elements[i] =              \
-                result.i##element_size##x##num_elements[num_elements - 1 - i]; \
-        }                                                                      \
-        PUT_V128_TO_ADDR(frame_lp + addr_ret, reversed_result);                \
-                                                                               \
-        break;                                                                 \
+#define SIMD_LOAD_OP(simde_func, element_size, num_elements) \
+    do {                                                     \
+        uint32 offset, addr;                                 \
+        offset = read_uint32(frame_ip);                      \
+        addr = GET_OPERAND(uint32, I32, 0);                  \
+        frame_ip += 2;                                       \
+        addr_ret = GET_OFFSET();                             \
+        CHECK_MEMORY_OVERFLOW(4);                            \
```

Comment:
doesn't it check the address of the v128 destination? if so, should that be 4 or rather 16?

Reply:
I'll check this once more